src/buildercore/bootstrap.py

```diff
@@ -229,34 +235,38 @@ def sub_sqs(stackname, context_sqs, region):
     """
     ensure(isinstance(context_sqs, dict), "Not a dictionary of queues pointing to their subscriptions: %s" % context_sqs)
 
-    sqs = core.boto_sqs_conn(region) # boto2
-    sns = core.boto_sns_conn(region) # boto2
+    sqs = core.boto_resource('sqs', region)
+    sns_client = core.boto_client('sns', region)
 
     for queue_name, subscriptions in context_sqs.items():
         LOG.info('Setup of SQS queue %s', queue_name, extra={'stackname': stackname})
-        queue = sqs.lookup(queue_name)
         ensure(isinstance(subscriptions, list), "Not a list of topics: %s" % subscriptions)
 
+        queue = sqs.Queue(queue_name)
         for topic_name in subscriptions:
             LOG.info('Subscribing %s to SNS topic %s', queue_name, topic_name, extra={'stackname': stackname})
 
             # idempotent, works as lookup
             # risky, may subscribe to a typo-filled topic name like 'aarticles'
             # there is no boto method to lookup a topic
-            topic_lookup = sns.create_topic(topic_name)
-            topic_arn = topic_lookup['CreateTopicResponse']['CreateTopicResult']['TopicArn']
+            topic_lookup = sns_client.create_topic(Name=topic_name) # idempotent
+            topic_arn = topic_lookup['TopicArn']
+
             # deals with both subscription and IAM policy
-            response = sns.subscribe_sqs_queue(topic_arn, queue)
-            assert 'SubscribeResponse' in response
-            assert 'SubscribeResult' in response['SubscribeResponse']
-            assert 'SubscriptionArn' in response['SubscribeResponse']['SubscribeResult']
-            subscription_arn = response['SubscribeResponse']['SubscribeResult']['SubscriptionArn']
-            # this method is not part of boto. see `buildercore.core._set_raw_subscription_attribute`
+            # http://boto.cloudhackers.com/en/latest/ref/sns.html#boto.sns.SNSConnection.subscribe_sqs_queue
+            # https://github.com/boto/boto/blob/develop/boto/sns/connection.py#L322
+            #response = sns.subscribe_sqs_queue(topic_arn, queue)
+            # WARN: doesn't do all of the above in boto3
+            #response = sns.subscribe(TopicArn=topic_arn, Protocol='sqs', Endpoint=queue.attributes['QueueArn'])
+            response = snsmod.subscribe_sqs_queue(sns_client, topic_arn, queue)
```

Comment:
@giorgiosironi , controversial change here. boto2 has a convenience function not present in boto3. It was straightforward to translate their code to boto3. I stuck it in a module called `sns.py` by itself. It's quite similar to your S3 policy wrangling by the looks of it :( 

I'm going to investigate the state of tests now for s3, everything keeps passing successfully which is making me suspicious.

Reply:
possibly this could return just the `SubscriptionARN`, to hide boto-specific details some more?